config_command.go

```diff
@@ -0,0 +1,182 @@
+package clio
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+
+	"github.com/anchore/fangs"
+)
+
+func ConfigCommand(app Application, options ...configCommandOption) *cobra.Command {
+	opts := configCommandOptions{}
+	for _, option := range options {
+		option(&opts)
+	}
+
+	id := app.ID()
+	internalApp := extractInternalApp(app)
+	if internalApp == nil {
+		return &cobra.Command{
+			RunE: func(_ *cobra.Command, _ []string) error {
+				return fmt.Errorf("unable to extract internal application, provided: %v", app)
+			},
+		}
+	}
+
+	cmd := &cobra.Command{
+		Use:   "config",
+		Short: fmt.Sprintf("show the %s configuration", id.Name),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			allConfigs := allCommandConfigs(internalApp)
+			var err error
+			if opts.loadConfig {
+				err = loadAllConfigs(cmd, internalApp.setupConfig.FangsConfig, allConfigs)
+			}
+			filter := opts.valueFilterFunc
+			if internalApp.state.RedactStore != nil {
+				filter = chainFilterFuncs(internalApp.state.RedactStore.RedactString, filter)
+			}
+			summary := summarizeConfig(cmd, internalApp.setupConfig.FangsConfig, filter, allConfigs)
+			_, writeErr := os.Stdout.WriteString(summary)
+			if writeErr != nil {
+				writeErr = fmt.Errorf("an error occurred writing configuration summary: %w", writeErr)
+				err = errors.Join(err, writeErr)
+			}
+			if err != nil {
+				// space before the error display
+				_, _ = os.Stderr.WriteString("\n")
+			}
+			return err
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.loadConfig, "load", "", opts.loadConfig, fmt.Sprintf("load and validate the %s configuration", id.Name))
+
+	if opts.includeLocationsSubcommand {
+		// sub-command to print expanded configuration file search locations
+		cmd.AddCommand(summarizeLocationsCommand(internalApp))
+	}
+
+	return cmd
+}
+
+type configCommandOption func(*configCommandOptions)
+
+type valueFilterFunc func(string) string
+
+type configCommandOptions struct {
+	loadConfig                 bool
+	includeLocationsSubcommand bool
+	valueFilterFunc            valueFilterFunc
+}
+
+// ReplaceHomeDirWithTilde adds a value filter function which replaces matching home directory values in strings
+// starting with the user's home directory to make configurations more portable
+func ReplaceHomeDirWithTilde(opts *configCommandOptions) {
```

Comment:
What about on the setup config that persists as unexported additional commands:
```golang
clioCfg := clio.NewSetupConfig(id).
 WithConfigCommand()
```

The upside is that this would be consistent with other options for a clio application. If a user wants to implement one themselves and add it they still can.

Reply:
This is similar to the [VersionCommand](https://github.com/anchore/clio/blob/main/version.go#L31). I think these should be consistent. If memory serves, we may not have had the root command tracking with `SetupRootCommand` and it was necessary to provide a separate way to get the version command added. Should we make both of these `WithVersionCommand(...)` and `WithConfigCommand(...)`?